dev-packages/cli/src/check-hoisting.ts

```diff
@@ -0,0 +1,145 @@
+/********************************************************************************
+ * Copyright (c) 2018-2019 TypeFox and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as fs from 'fs';
+import * as path from 'path';
+
+/**
+ * This script makes sure all the dependencies are hoisted into the root `node_modules` after running `yarn`.
+ *  - https://github.com/theia-ide/theia/pull/2994#issuecomment-425447650
+ *  - https://github.com/theia-ide/theia/pull/2994#issuecomment-425649817
+ *
+ * If you do not want to bail the execution, set the `THEIA_CHECK_HOISTING_NO_BAIL` environment variable to `true`. Use `cross-env`.
+ * ```json
+ *   ...
+ *   "scripts": {
+ *     ...,
+ *     "prepare:hoisting": "cross-env THEIA_CHECK_HOISTING_NO_BAIL=true node scripts/check-hoisting",
```

Comment:
Please keep this line in sync with https://github.com/theia-ide/theia/pull/4138/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R51.

Also, maybe we could remove `cross-env` from the main `package.json` and move it to either the `@theia/application-manager` or the `@theia/cli`. Thoughts?

Reply:
@kittaakos i was thinking to replace it with cli option, WDYT? something like `theia check:hoisted --silent` then we don't need `cross-env`

Reply:
Sure, your approach is even better.

Reply:
@kittaakos I've went with `suppress` option, no `silent`. Good to merge like that?